src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs

```diff
@@ -62,13 +62,13 @@ public ConfigurationSetting(string key, string value, string label = null)
         /// <summary>
         /// The last time a modifying operation was performed on the given key-value.
         /// </summary>
-        public DateTimeOffset LastModified { get; set; }
+        public DateTimeOffset? LastModified { get; set; }
 
         /// <summary>
         /// A value indicating whether the key-value is locked.
         /// A locked key-value may not be modified until it is unlocked.
         /// </summary>
-        public bool Locked { get; set; }
+        public bool? Locked { get; set; }
```

Comment:
Does this need to be get-set? i.e. should it be a get only property? Same for the LastModified above.

Reply:
Sure thing. Changed

Reply:
I don't agree with @KrzysztofCwalina here. We shouldn't artificially limit access to members of data types as they might be used in customer unit tests along with mocked clients and making setters internal wouldn't allow testing some scenarios.

Reply:
For this specific data types, the user shouldn't be able to update the values. These are only set by the server, and making local changes and then submit them to the server could give a false idea that they are able to overwrite them.
Avoiding the setters sounds like the best approach

Reply:
How would customer be able to write a test like this:

https://github.com/aspnet/Extensions/blob/master/src/Configuration/Config.AzureKeyVault/test/AzureKeyVaultConfigurationTest.cs#L104

or this:

https://github.com/aspnet/Extensions/blob/master/src/Configuration/Config.AzureKeyVault/test/AzureKeyVaultConfigurationTest.cs#L146

If they are not able to set the properties withouth using reflection.